k2/csrc/properties_test.cc

```diff
@@ -356,5 +357,41 @@ TEST(Properties, IsConnected) {
     EXPECT_TRUE(is_connected);
   }
 }
+TEST(FsaAlgo, IsAcyclic) {
+  // empty fsa is acyclic
+  {
+    Fsa fsa;
+    bool is_acyclic = IsAcyclic(fsa);
+    EXPECT_TRUE(is_acyclic);
+  }
+
+  // an acyclic fsa example
+  {
+    std::vector<Arc> arcs = {
+      {0, 1, 2}, {0, 2, 1}, {1, 2, 0}, {1, 3, 5}, {2, 3, 6},
+    };
+    std::vector<int32_t> arc_indexes = {0, 2, 4, 5};
```

Comment:
We have add a constructor only with `arcs`

https://github.com/danpovey/k2/blob/master/k2/csrc/fsa.h#L93

so we can write now `Fsa fsa(std::move(arcs), 3)`